### DISCUSSION TITLE
How to handle "label does not exist" when building multi-documents?

### DISCUSSION THREAD

**Discussion by Holt59** [typst/typst#733] [Q&A]
I am working on a document that I separated in various file and then use a `main.typ` that includes other files. When I build `main.typ`, there are no issues with labels and reference but I try to build file separately, I get error when referencing labels from other files.

Is there a way to just "skip" these errors and output some nonsense (e.g., the raw "@xxx") with a simple warning? I'd like to be able to compile the various part separately.

**Comment by laurmaedje** [comment]
Do these errors occur when compiling with the command line compiler / web app or do they only exist with the Typst LSP?

**Reply by Holt59** [reply]
> Do these errors occur when compiling with the command line compiler / web app or do they only exist with the Typst LSP?

These occurs with both Typst LSP and the command line compiler.

**Reply by laurmaedje** [reply]
Why do you want to compile the individual files separately? Are you sure that is the right approach?

**Reply by Holt59** [reply]
One of the reason is that the VS Code extension does not detect change in included file as far as I know, so having to manually compile the main file every time one of the included file is modified is a pain.

Second reason is that with large latex document, it is common to only want to compile part of the document. I know that typst compilation are way faster than latex ones, but it's very useful to be able to send part of the document without having to play with commented include, etc.

The label are the only thing that prevent the per-file compilation right now, everything else is working fine.

**Comment by ondohotola** [comment]
@Holt59 See issue #1276 for the first step, I have also raised now #1322 as consequential.

@laurmaedje I agree with @Holt59 I also often only compile an `#include#` document because sometimes a hardcopy is useful.
